sdk/python/kubeflow/storage_init_container/hugging_face.py

```diff
@@ -57,3 +59,37 @@ def download_model_and_tokenizer(self):
         transformers.AutoTokenizer.from_pretrained(
             self.model, cache_dir=self.config.download_dir
         )
+
+
+@dataclass
+class HfDatasetParams:
+    repo_id: str
+    access_token: str = None
+    allow_patterns: list[str] = None
+    ignore_patterns: list[str] = None
+    download_dir: str = field(default="/workspace/datasets")
+
+    def __post_init__(self):
+        # Custom checks or validations can be added here
+        if self.repo_id is None:
+            raise ValueError("repo_id is None")
+
+
+class HuggingFaceDataset(datasetProvider):
+    def load_config(self, serialised_args):
+        self.config = HfDatasetParams(**json.loads(serialised_args))
+
+    def download_dataset(self):
+        print("downloading dataset")
```

Comment:
Would it be hard to use logging here instead of print ?

Reply:
Can I take care of logging part once the entire api is running end to end?

Reply:
Sure, let's do it after!